### PR TITLE
Emxd 3336 add app video ctv

### PR DESCRIPTION
**Pull request by EMXDigital**
Adding video support to EMX Digital adapter

### CODE REVIEW

**Comment by EMXDigital** [comment]
Juts following up here, the checks seem to be failing due to an independent/non-related part of the package `pubstack_module_test`

` FAIL: TestPubstackModuleSuccess (0.17s)
    pubstack_module_test.go:223: 
        	Error Trace:	pubstack_module_test.go:223
        	            				pubstack_module_test.go:205
        	Error:      	Should receive an event, but did NOT
        	Test:       	TestPubstackModuleSuccess
        	Messages:   	setUID events are only published when logging a setUID object with setUID feature on
FAIL`

EMX Digital adapter looks to be okay and has > 90% testing coverage

**Comment by bsardo** [comment]
I'll take a look at that test failure since I recently worked on that test.

**Comment by EMXDigital** [comment]
@bsardo looks like all the checks are passing now 👍

File: adapters/emx_digital/emx_digital.go [hunk 1]
```diff
@@ -137,6 +137,41 @@ func buildImpBanner(imp *openrtb.Imp) error {
...
+	if imp.Video.Protocols != nil {
+		imp.Video.Protocols = cleanProtocol(imp.Video.Protocols)
+	}
+
+	return nil
+}
+
+// not supporting VAST protocol 7 (VAST 4.0);
+func cleanProtocol(protocols []openrtb.Protocol) []openrtb.Protocol {
+	newitems := []openrtb.Protocol{}
```
Comment (SyntaxNode) [inline adapters/emx_digital/emx_digital.go:164]: Consider using make to specify an initial capacity. It's slightly better memory management:
```
newitems := make([]openrtb.Protocol{}, 0, len(protocols))
```

File: adapters/emx_digital/emx_digital.go [hunk 2]
```diff
@@ -137,6 +137,41 @@ func buildImpBanner(imp *openrtb.Imp) error {
 	return nil
 }
 
+func buildImpVideo(imp *openrtb.Imp) error {
+	imp.Ext = nil
```
Comment (SyntaxNode) [inline adapters/emx_digital/emx_digital.go:141]: Just curious why the Ext is removed here.

**Review by SyntaxNode — COMMENTED** [review]
Looks good. I have a few minor comments. Please review the docs for this adapter to check if any changes are needed: https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders/emx_digital.md

File: adapters/emx_digital/emx_digital.go (hunk 1 above)
Comment (EMXDigital) [reply adapters/emx_digital/emx_digital.go:164]: Makes sense to us, appreciate the suggestion

File: adapters/emx_digital/emx_digital.go (hunk 2 above)
Comment (EMXDigital) [reply adapters/emx_digital/emx_digital.go]: Team had removed it as we were not supporting it for display either - now adapter supports for both display and video for future use

**Comment by EMXDigital** [comment]
@SyntaxNode Updated code based on feedback and discussion with team. Ready for another round

**Comment by SyntaxNode** [comment]
Is the intent to support the `/openrtb2/video` long form video endpoint? If so, please look at the "Long-Form Video Support" section of the [new adapter docs](https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html) to be sure you are setting the necessary fields.

**Comment by EMXDigital** [comment]
@SyntaxNode No, we are not looking to support long form video at this time, just the vanilla video integrations

**Comment by SyntaxNode** [comment]
> @SyntaxNode No, we are not looking to support long form video at this time, just the vanilla video integrations

Gotcha. Sometimes CTV = Long Form Video, so I wanted to ask. Thank you.

File: adapters/emx_digital/emx_digital.go [hunk 3]
```diff
@@ -170,14 +203,22 @@ func preprocess(request *openrtb.BidRequest) []error {
...
 	if request.Site != nil && request.Site.Page != "" {
-		pageURL, err := url.Parse(request.Site.Page)
-		if err == nil && pageURL.Scheme == "https" {
-			secure = int8(1)
+		domain = request.Site.Page
+	} else if request.App != nil {
+		if request.App.Domain != "" {
+			domain = request.App.Domain
+		} else if request.App.StoreURL != "" {
+			domain = request.App.StoreURL
```
Comment (SyntaxNode) [inline adapters/emx_digital/emx_digital.go:213]: Can you please add test coverage for the `StoreURL` condition?

File: adapters/emx_digital/emx_digital.go [hunk 4]
```diff
@@ -238,6 +286,19 @@ func (a *EmxDigitalAdapter) MakeBids(internalRequest *openrtb.BidRequest, extern
...
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Video != nil {
+				bidType = openrtb_ext.BidTypeVideo
+			}
+			break
+		}
+	}
+	return bidType
+}
```
Comment (SyntaxNode) [inline adapters/emx_digital/emx_digital.go:300]: Does your bid response include the bid type? Although the code you have here is common in the project, there's been issues reported (see https://github.com/prebid/prebid-server/issues/862) that this approach isn't always reliable for multiformat impressions. Is there enough information to detect the type based on the bid response?

Comment (SyntaxNode) [reply adapters/emx_digital/emx_digital.go:306]: Thank you very much for the fix here.

**Review by SyntaxNode — APPROVED** [review]

**Review by hhhjort — APPROVED** [review]

**Comment by EMXDigital** [comment]
Thanks @hhhjort for merging this in - will there be an official release coming out in the coming days that we can use to message to Publishers for what version of Prebid S2S supports this new functionality?

**Comment by hhhjort** [comment]
We don't have a set in stone schedule for releases, but there is likely to be a release tomorrow. If not then there should be a release in the middle of next week. The version will be 0.134.0 of course.

**Pull request merged**